_posts/2023-09-25-Debezium-JMX-signaling-and-notifications.adoc

```diff
@@ -0,0 +1,1558 @@
+    "signal.data.collection": "inventory.debezium_signal",
+    "notification.enabled.channels": "jmx" <1>
+  }
+}
+----
+<1> This configuration enables **jmx** notification channel.
+
+To access the notifications we need to connect again to JMX server. So as we did for the signal we will use `jmxterm`
+
+[source,bash]
+----
+java -jar jmxterm-1.0.4-uber.jar <1>
+
+open localhost:9012 <2>
+
+beans -d debezium.postgres <3>
+
+get -b debezium.postgres:context=notifications,server=dbserver1,type=management Notifications <4>
+----
+<1> Run the jmxterm client
+<2> Open a connection to JMX Server
+<3> Search for bean under **debezium.postgres** domain
+<4> Get notifications.
+
+you should expect the following output
+
+[source, text]
+----
+#mbean = debezium.postgres:context=notifications,server=dbserver1,type=management:
+Notifications = [ {
+  additionalData = {
+    ( connector_name ) = {
+      key = connector_name;
+      value = dbserver1;
+     };
+   };
+  aggregateType = Initial Snapshot;
+  id = b20bec8d-f21f-4d74-bb75-cdd7f4c7d933;
+  type = SKIPPED;
+ }, <1>
+{
+  additionalData = {
+    ( connector_name ) = {
+      key = connector_name;
+      value = dbserver1;
+     };
+    ( data_collections ) = {
+      key = data_collections;
+      value = inventory.orders;
+     };
+   };
+  aggregateType = Incremental Snapshot;
+  id = 12345;
+  type = STARTED;
+ }, <2>
+{
+  additionalData = {
+    ( current_collection_in_progress ) = {
+      key = current_collection_in_progress;
+      value = inventory.orders;
+     };
+    ( connector_name ) = {
+      key = connector_name;
+      value = dbserver1;
+     };
+    ( maximum_key ) = {
+      key = maximum_key;
+      value = 10004;
+     };
+    ( last_processed_key ) = {
+      key = last_processed_key;
+      value = 10004;
+     };
+    ( data_collections ) = {
+      key = data_collections;
+      value = inventory.orders;
+     };
+   };
+  aggregateType = Incremental Snapshot;
+  id = 12345;
+  type = IN_PROGRESS;
+ }, <3>
+{
+  additionalData = {
+    ( scanned_collection ) = {
+      key = scanned_collection;
+      value = inventory.orders;
+     };
+    ( connector_name ) = {
+      key = connector_name;
+      value = dbserver1;
+     };
+    ( total_rows_scanned ) = {
+      key = total_rows_scanned;
+      value = 4;
+     };
+    ( status ) = {
+      key = status;
+      value = SUCCEEDED;
+     };
+    ( data_collections ) = {
+      key = data_collections;
+      value = inventory.orders;
+     };
+   };
+  aggregateType = Incremental Snapshot;
+  id = 12345;
+  type = TABLE_SCAN_COMPLETED;
+ }, <4>
+{
+  additionalData = {
+    ( connector_name ) = {
+      key = connector_name;
+      value = dbserver1;
+     };
+   };
+  aggregateType = Incremental Snapshot;
+  id = 12345;
+  type = COMPLETED;
+ } <5>
+];
+----
+<1> This is a notification from initial snapshot that as the status `SKIPPED` since our connector is configured with `"snapshot.mode": "NEVER"`
```

Comment:
```suggestion
<1> This is a notification from the initial snapshot with the status `SKIPPED` since our connector is configured with `"snapshot.mode": "NEVER"`
```